### PR TITLE
Read More link enhancements

### PR DESCRIPTION
**Pull request by markbaird**
This includes the changes to "read more" links we discussed in #68.

### CODE REVIEW

File: include/service/entities/content/article_service_v2.js [hunk 1]
```diff
@@ -145,23 +145,54 @@ module.exports = function(pb) {
...
+        if (util.isFunction(options)) {
+            cb      = options;
+            options = {};
+        }
+        if (options.readMore === undefined) {
+            options.readMore = false;
+        }
+
+        ArticleServiceV2.super_.prototype.getSingle.apply(this, [options, cb]);
+    };
```
Comment (brianhyder) [inline include/service/entities/content/article_service_v2.js]: Good catch.

File: include/service/entities/content/article_service_v2.js [hunk 2]
```diff
@@ -145,23 +145,54 @@ module.exports = function(pb) {
...
+            self.render([article], renderOptions, function(err/*, articles*/) {
                 cb(err, article);
             });
         };
         ArticleServiceV2.super_.prototype.get.apply(this, [id, options, afterGet]);
     };
+
+    /**
+     *
+     * @method getAll
```
Comment (brianhyder) [inline include/service/entities/content/article_service_v2.js]: Can you update the name to ```getSingle```

File: include/service/entities/content/article_service_v2.js [hunk 3]
```diff
@@ -145,23 +145,54 @@ module.exports = function(pb) {
...
      * @param {object} options
      * @param {Boolean} [options.render=false]
+     * @param {Boolean} [options.readMore=false]
      * @param {Function} cb
      */
     ArticleServiceV2.prototype.get = function(id, options, cb) {
-        
         var self = this;
+        var renderOptions = {};
+        if (!options || options.readMore === undefined) {
```
Comment (brianhyder) [inline include/service/entities/content/article_service_v2.js]: I think you can simplify the logic here by doing:
```
var renderOptions = {
  readMore: options && options.readMore ? true : false
};
```

File: include/service/entities/content/article_service_v2.js [hunk 4]
```diff
@@ -145,23 +145,54 @@ module.exports = function(pb) {
...
+        }
+
         var afterGet = function(err, article) {
             if (util.isError(err) || article === null || !options || !options.render) {
                 return cb(err, article);
             }
             
             //complete the rendering
-            self.render([article], function(err/*, articles*/) {
+            self.render([article], renderOptions, function(err/*, articles*/) {
```
Comment (brianhyder) [inline include/service/entities/content/article_service_v2.js]: One small nit-picky item.  Would you mind moving the creation of the ```renderOptions``` block here?  It marries the logic the options to the code that utilizes it.

File: test/include/service/entities/content/article_renderer_tests.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
+
+//depedencies
+var should          = require('should');
+var UrlService      = require('../../../../../include/service/entities/url_service.js')({config: {siteRoot: 'http://www.test.com'}});
+var ArticleRenderer = require('../../../../../include/service/entities/content/article_renderer.js')({UrlService: UrlService});
```
Comment (brianhyder) [inline test/include/service/entities/content/article_renderer_tests.js]: These tests are awesome!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by brianhyder** [comment]
@markbaird this is great.  Would you also mind looking at the section, article, and blog controllers to make sure they are passing the new option correctly?

**Comment by markbaird** [comment]
@brianhyder I already tested the blog controller and it is working correctly. I'll make the changes you noted and test those controllers later today.

**Comment by markbaird** [comment]
@brianhyder also, is Coveralls not working anymore? I wanted to see if the new tests increased the coverage any.

File: plugins/pencilblue/controllers/section.js [hunk 6]
```diff
@@ -120,7 +120,7 @@ module.exports = function(pb) {
                 order: [{'publish_date': pb.DAO.DESC}, {'created': pb.DAO.DESC}]
             };
             pb.ArticleServiceV2.setPublishedClause(opts.where);
-            self.service.getBySectionViewController(section, opts, function(err, content) {
+            self.service.getBySection(section, opts, function(err, content) {
```
Comment (markbaird) [inline plugins/pencilblue/controllers/section.js]: This was throwing an exception because getBySectionViewController() doesn't exist.

**Comment by markbaird** [comment]
I implemented your suggestions and tested the blog, article and section controllers again. The section controller actually had a typo causing an exception to be thrown. I checked that I hadn't introduced the bug by verifying it was present on master, then I fixed it and included the fix in this pull request.

File: plugins/pencilblue/controllers/section.js (hunk 6 above)
Comment (brianhyder) [inline plugins/pencilblue/controllers/section.js]: Fail on my part.  Thank you for correcting.

**Pull request merged**